src/device/wg_controller.rs

```diff
@@ -151,8 +151,11 @@ async fn consolidate_wg_peers<
     post_quantum_vpn: &impl telio_pq::PostQuantum,
     features: &Features,
 ) -> Result {
-    let proxy_endpoints = if let Some(p) = proxy {
-        p.get_endpoint_map().await?
+    let (proxy_endpoints, replaced_proxy_endpoints) = if let Some(p) = proxy {
```

Comment:
replace proxy endpoints is not used in is_peer_proxying, 

Maybe we should just replace get_endpoints to return:
HashMap<PublicKey, Vec<SocketAddr>>, assuming first is the one that should be used, and additional are for doing x.contains(sock) for places during verification ?